cloud/src/meta-service/meta_service_resource.cpp

```diff
@@ -510,13 +510,115 @@ static void set_default_vault_log_helper(const InstanceInfoPB& instance,
     LOG(INFO) << vault_msg;
 }
 
+static int alter_hdfs_storage_vault(InstanceInfoPB& instance, std::unique_ptr<Transaction> txn,
+                                    const StorageVaultPB& vault, MetaServiceCode& code,
+                                    std::string& msg) {
+    if (!vault.has_hdfs_info()) {
```

Comment:
The following code would do the job.
```C++
const auto& hdfs_info = vault.hdfs_info();
    if (hdfs_info.has_prefix() || !hdfs_info.has_build_conf() ||
        hdfs_info.build_conf().has_fs_name()) {
        code = MetaServiceCode::INVALID_ARGUMENT;
        std::stringstream ss;
        ss << "You can not alter prefix or fs name because it might lose previoud written data";
        msg = ss.str();
        return -1;
    }
```